app.js

```diff
@@ -1,77 +1,164 @@
 var leftEl = document.getElementById('left');
 var centerEl = document.getElementById('center');
 var rightEl = document.getElementById('right');
+var clicks = document.getElementById('clicks');
 
-var img = [
-  './img/bag.jpg', './img/banana.jpg', './img/bathroom.jpg', './img/boots.jpg', './img/breakfast.jpg', './img/tauntaun.jpg', './img/wine-glass.jpg', './img/bubblegum.jpg', './img/chair.jpg', './img/cthulhu.jpg', './img/dog-duck.jpg', './img/dragon.jpg', './img/unicorn.jpg', './img/usb.gif', './img/pen.jpg', './img/pet-sweep.jpg', './img/scissors.jpg', './img/shark.jpg', './img/sweep.png', './img/water-can.jpg'];
-
+var allProducts = [];
 var totalClicks = 0;
 var clickLimits = 25;
-
-function product(name){
+var previousImgs = [];
+
+var productNames = [
+  { name: 'bag', path: './img/bag.jpg' },
+  { name: 'banana', path: './img/banana.jpg' },
+  { name: 'bathroom', path: './img/bathroom.jpg'},
+  { name: 'boots', path: './img/boots.jpg'},
+  { name: 'breakfast', path: './img/breakfast.jpg'},
+  { name: 'tauntaun', path: './img/tauntaun.jpg'},
+  { name: 'wine-glass', path: './img/wine-glass.jpg'},
+  { name: 'bubblegum', path: './img/bubblegum.jpg'},
+  { name: 'chair', path: './img/chair.jpg'},
+  { name: 'cthulhu', path: './img/cthulhu.jpg'},
+  { name: 'dog-duck', path: './img/dog-duck.jpg'},
+  { name: 'dragon', path: './img/dragon.jpg'},
+  { name: 'unicorn', path: './img/unicorn.jpg'},
+  { name: 'usb', path: './img/usb.gif'},
+  { name: 'pen', path: './img/pen.jpg'},
+  { name: 'pet-sweep', path: './img/pet-sweep.jpg'},
+  { name: 'scissors', path: './img/scissors.jpg'},
+  { name: 'shark', path: './img/shark.jpg'},
+  { name: 'sweep', path: './img/sweep.png'},
+  { name: 'water-can', path: './img/water-can.jpg'}
+];
+
+//product constructor
+function Product(name, path){
   this.name = name;
   this.views = 0;
   this.clicks = 0;
-  //this.path = 'img/' + name + jpg';
+  this.path = path;
 };
+//create function for all products & push to their array
+function createProducts(){
+  for (var i = 0; i < productNames.length; i++) {
+    var productInfo = productNames[i];
+    var newItem = new Product(productInfo.name, productInfo.path);
+    allProducts.push(newItem);
+  }
+  console.log('allProducts: ', allProducts);
+}
 
-var previousImgs = [leftIndex, centerIndex, rightIndex];
-
-function randNum(){
-  return Math.floor(Math.floor(Math.random() * img.length));
+function randomNum(){
+  return Math.floor(Math.random() * productNames.length);
 };
 
-/*function displayPics() {
+function displayFinalResults() {
+  console.log('displayFinalResults');
+  for (var i = 0; i < allProducts.length; i++) {
+    var productClicksEl = document.createElement('li');
+    console.dir(productClicksEl);
+    clicks.appendChild(productClicksEl);
+    productClicksEl.textContent = allProducts[i].name + ': clicks ' + allProducts[i].clicks + ', views ' + allProducts[i].views;
+  }
+}
+
+function generateImg() {
```

Comment:
Is the process of generating a random number for the left image much different than the process of generating it for the right one? Can you think of a way to use a loop to repeat that task instead of hardcoding it 3 times?